src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -375,25 +431,110 @@ def check_token_status(self, clustersTokenMap=None):
+            if '.ceph-dashboard' in prometheus_url:
+                prometheus_url = prometheus_url.replace('.ceph-dashboard', '')
+            parsed_url = urlparse(prometheus_url)
+            scheme = parsed_url.scheme
+            hostname = parsed_url.hostname
+            try:
+                # Check if the hostname is already an IP address
+                ipaddress.ip_address(hostname)
+                valid_ip_url = True
+            except ValueError:
+                valid_ip_url = False
+
+            orch = OrchClient.instance()
+            inventory_hosts = (
+                [host.to_json() for host in orch.hosts.list()]
+                if not valid_ip_url
+                else []
+            )
+
+            def find_node_ip():
                 for host in inventory_hosts:
-                    if host['hostname'] == hostname or host['hostname'] in hostname:
-                        node_ip = host['addr']
-                prometheus_url = prometheus_url.replace(hostname, node_ip)
+                    if host['hostname'] == hostname or hostname in host['hostname']:
+                        return host['addr']
+                return None
+
+            node_ip = find_node_ip() if not valid_ip_url else None
+            prometheus_url = prometheus_url.replace(hostname, node_ip) if node_ip else prometheus_url  # noqa E501 #pylint: disable=line-too-long
+            if mgmt_gw_enabled:
+                prometheus_url = f"{scheme}://{node_ip if node_ip else hostname}"
         return prometheus_url
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any],
+                                    target: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster.get('prometheus_url')
+                if prometheus_url:
+                    endpoint = (
+                        prometheus_url.replace("https://", "").replace("http://", "")
+                    )  # since target URLs are without scheme
+
+                    if endpoint == target:
+                        return cluster.get('prometheus_access_info')
+        return None
+
+    def get_cluster_credentials(self, targets: List[str]) -> Dict[str, Any]:
+        clusters_credentials: Dict[str, Dict[str, Any]] = {}
+        multi_cluster_config = self.load_multi_cluster_config()
+
+        # Return early if no multi_cluster_config is loaded
+        if not multi_cluster_config:
+            return clusters_credentials
+
+        try:
+            for target in targets:
+                credentials = self.find_prometheus_credentials(multi_cluster_config, target)
+                if credentials:
+                    clusters_credentials[target] = credentials
+                    clusters_credentials[target]['cert_file_name'] = ''
+                else:
+                    logger.error('Credentials not found for target: %s', target)
```

Comment:
So? Is the used expected to do something else or is an unrecoverable error? Please, provide help on how to fix the issue.